app.ts

```diff
@@ -0,0 +1,183 @@
+			const lastRow = $(".row:last");
+			for (let i = 0; i < records[r].length; i++) {
+				lastRow.append(`<td>${records[r][i]}</td>`); //assign each record to their column in a specified row
+			}
+			if (records[r].includes(inputValue)) {
+				lastRow.css('background-color', '#DDC0B4'); // hightlights the searched row
+			}
+			tbody.append(lastRow);
+		}
+	} catch (error) {
+		console.error("Error displaying records:", error);
+	}
+}
+displayRecords()
+
+async function updateRecordsAndResize(inputValue: number) {
+	let count = await fetchRecordCount() - 1;
+	if (inputValue < 0 || inputValue > count) { // check if the search input 
+		$('.modal').css('display', 'block') //opens modal if search input is not within range
+		$('.modal-content').append(`<p>${inputValue} is not a number within the range.Please try a different number</p>`)
+		$('#searchInput').val(''); // empties search bar
+		return;
+	}
+	let calculatedRows = adjustRowsByScreenHeight();
+	const quarterRange = Math.floor(calculatedRows / 2); // divides the calculated max rows in half 
+	firstNumber = Math.max(0, inputValue - quarterRange);
+	lastNumber = Math.min(count, firstNumber + (calculatedRows - 1));
+	await displayRecords();
+}
+
+$('#closeModalBtn').on("click", () => {
+	$('.modal').css('display', 'none') // closes modal
+});
+
+$('.btnSearch').on('click', async (event: any) => {
+	event.preventDefault();
+	let inputValue = $('#searchInput').val() as number;
+	await updateRecordsAndResize(inputValue); // calls to calculate the range once button is clicked
+});
+
+async function rightArrow(): Promise<void> {
+	$('#searchInput').val('')
+	const lastRow = document.querySelector("#recordsTable tbody .row:last-child"); // retrieves the last row
+	let count = await fetchRecordCount() - 1;
+	if (lastRow) { // checks if the last row exists
+		const cells = lastRow.querySelectorAll("td");
+		const lastRecord: string[] = [];
+		cells.forEach((cell) => {
+			lastRecord.push(cell.textContent || "");
+		});
+		const lastID = parseFloat(lastRecord[0]); // determines te value in the last row
+		if (0 <= lastID && lastID <= (count)) { // checks if the last value is within range
+			const tbody = $("tbody");
+			tbody.empty(); // empties the table
+			firstNumber = lastID + 1; // calculates the first number of the page
+			let calculatedRows = await adjustRowsByScreenHeight();
+			lastNumber = firstNumber + (calculatedRows - 1) // calculates the first number of the page
+			await displayRecords(); // display the new records
+		}
+	}
+}
+
+async function leftArrow(): Promise<void> {
+	$('#searchInput').val('')
+	let count = await fetchRecordCount() - 1;
+	const firstRow = document.querySelector("#recordsTable tbody .row:first-child"); // retrieves the first row
+	if (firstRow) { // checks if the first row exists
+		const cells = firstRow.querySelectorAll("td");
+		const firstRecord: string[] = [];
+		cells.forEach((cell) => {
```

Comment:
Rather make use of a `for..of` loop, even a for loop is fine, but for of is preferred